### DISCUSSION TITLE
Best upgrade path

### DISCUSSION THREAD

**Discussion by emmarichardson** [keycloak/keycloak#32743] [Q&A]
Currently running Keycloak 20.0.1 on an Ovirt VM.  Been slacking off (if it ain't broke philosophy) and realized that I am now several versions behind and really should get it upgraded.  Just wondering what is the best path -
Can I upgrade directly to 25?  Or what interim steps do I need to go through..I would love it if the upgrade documentations would have required versions documentation.
Am I better off to just fire up a new server and migrate the realm across (only have one) or is this a more problematic path?
If upgrading, my thought is just to hit this over the weekend when there is very little use and snapshot as I go..
I have several oauth and saml clients, a couple of ldap servers, and one plugin.  Anything thing I should be aware of - I saw there was some saml changes but didn't quite understand if that is going to break something in standard saml clients or not...
Would appreciate any help.

**Comment by jonkoops** [comment]
Best option is to upgrade to the latest patch version major by major. Since 21 has a minor version as well I'd recommend going to [21.1.2](https://github.com/keycloak/keycloak/releases/tag/21.1.2), [22.0.5](https://github.com/keycloak/keycloak/releases/tag/22.0.5), [23.0.7](https://github.com/keycloak/keycloak/releases/tag/23.0.7), [24.0.5](https://github.com/keycloak/keycloak/releases/tag/24.0.5) and finally [25.0.5](https://github.com/keycloak/keycloak/releases/tag/25.0.5). I would recommend you always do a full backup of your database at each step.
